sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -3131,7 +3131,7 @@ namespace Azure { namespace Storage { namespace Blobs {
 
       static Azure::Core::Http::Request SetAccessTierConstructRequest(
           const std::string& url,
-          const SetAccessTierOptions& options)
+          SetAccessTierOptions& options)
```

Comment:
@JinmingHu-MSFT ,  all options can still be `const`.  Why did we remove it?

Reply:
If it was `const`, how can we move that `std::unique_ptr`?

Reply:
I would let body out of the options and as an extra argument for Constructing the Request, just like is it now an extra arg for some other functions.

Reply:
@vhvb1989 I'm also okay with this way.

But there's this conclusion that Azure Core team and Azure Storage team and Jeffrey all agreed on in the meeting months ago, that in protocol layer interface, there're only two parameters, url and options.

I'm not sure if we can just overturn that decision.

Reply:
yeah, I remember.
However, we didn't consider having a unique_ptr in the options (that needs to be moved).
I also find kind of weird to have a body as an option.  It has a benefit for non-body requests like GET/HEAD, but we could easily use nullptr as arg.

Having `cost` for options IMO is the best wat to make sure options won't be mutated. So I wouldn't change that.

I would like to bring this to our meeting @gilbertw , @JeffreyRichter 

As for now, let's use options without `const` as the last agreement.
Thank you @JinmingHu-MSFT

Reply:
@vhvb1989 Let's bring this up in the weekly meeting, so that we can make a decision together. Once we have a conclusion, i'm going to change the code. But before that, let's just keep it as it is now.